arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Ior.kt

```diff
@@ -144,7 +144,7 @@ public sealed class Ior<out A, out B> {
    * @param fab the function to apply if this is a [Both]
    * @return the results of applying the function
    */
-  public inline fun <C> fold(fa: (A) -> C, fb: (B) -> C, fab: (A, B) -> C): C {
+  public inline fun <C> fold(crossinline fa: (A) -> C, crossinline fb: (B) -> C, crossinline fab: (A, B) -> C): C {
```

Comment:
Same here. I think it's useful to allow early-returns here. Is the K2 compiler warning about this somehow?

Reply:
Sorry for the noise, this was indeed a leftover of my initial attempt to remove some "leaked function" errors. At the end, the solution there was to manually rewrite some functions, because the compiler is not clever enough to see "through" other inline functions.